xarray/indexes/interval_index.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from collections.abc import Hashable, Iterable, Mapping, Sequence
+from typing import TYPE_CHECKING, Any, cast
+
+import numpy as np
+import pandas as pd
+
+from xarray import Variable
+from xarray.core.indexes import Index, PandasIndex
+from xarray.core.indexing import IndexSelResult
+
+if TYPE_CHECKING:
+    from xarray.core.types import Self
+
+
+class IntervalIndex(Index):
+    """Xarray index of 1-dimensional intervals.
+
+    This index is built on top of :py:class:`~xarray.indexes.PandasIndex` and
+    wraps a :py:class:`pandas.IntervalIndex`. It is associated with two
+    coordinate variables:
+
+    - a 1-dimensional coordinate where each label represents an interval that is
+      materialized by its midpoint (i.e., the average of its left and right
+      boundaries)
```

Comment:
```suggestion
      materialized by a central value (commonly the average of its left and right
      boundaries)
```

This is the cool bit! Otherwise we wouldn't need it